### PR TITLE
Deprecate the `--isolated` flag

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

This PR deprecates the `--isolated` flag. The treatment varies across the APIs:

- For non-preview APIs, we warn but treat it as equivalent to `--no-config`.
- For preview APIs, we warn and ignore it, with two exceptions...
- For `tool run` and `run` specifically, we don't even warn, because we can't differentiate the command-specific `--isolated` from the global `--isolated`.

### CODE REVIEW

**Comment by j178** [comment]
Could you also update the `sync-python-releases.yml` workflow and the module commets in the `uv-python/*.py` scripts? Or do you think it would be better to handle this in a separate PR?

**Comment by charliermarsh** [comment]
Yes, thanks for the reminder.

File: scripts/release.sh [hunk 1]
```diff
@@ -11,14 +11,14 @@ echo "Updating metadata with rooster..."
...
-uv tool run --from 'rooster-blue>=0.0.7' --python 3.12 --isolated -- \
+uv tool run --from 'rooster-blue>=0.0.7' --python 3.12 --isolate -- \
     rooster release "$@" \
     --only-sections preview \
     --changelog-file PREVIEW-CHANGELOG.md \
     --no-update-pyproject --no-update-version-files
 
 # Update the real changelog
-uv tool run --from 'rooster-blue>=0.0.7' --python 3.12 --isolated -- \
+uv tool run --from 'rooster-blue>=0.0.7' --python 3.12 --isolate -- \
```
Comment (charliermarsh) [inline scripts/release.sh:21]: This is probably not necessary?

File: .github/workflows/sync-python-releases.yml [hunk 2]
```diff
@@ -20,8 +20,8 @@ jobs:
       - uses: hynek/setup-cached-uv@v1
       - name: Sync Python Releases
         run: |
-          uv run --isolated -- fetch-download-metadata.py
-          uv run --isolated -- template-download-metadata.py
+          uv run --no-workspace -- fetch-download-metadata.py
+          uv run --no-workspace -- template-download-metadata.py
```
Comment (charliermarsh) [inline .github/workflows/sync-python-releases.yml:24]: A little tedious...

**Comment by charliermarsh** [comment]
@zanieb - I will update docs in this PR once the stack is approved, since it's possible that we'll iterate on the names.

File: .github/workflows/sync-python-releases.yml (hunk 2 above)
Comment (konstin) [reply .github/workflows/sync-python-releases.yml:24]: Doesn't inline metadata imply isolation? `uv run fetch-download-metadata.py` works for me

**Review by konstin — APPROVED** [review]

File: crates/uv-python/fetch-download-metadata.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
 Usage:
 
-    uv run --isolated -- crates/uv-python/fetch-download-metadata.py
+    uv run --no-workspace -- crates/uv-python/fetch-download-metadata.py
```
[the remark this answers is not in the document]
Comment (charliermarsh) [reply crates/uv-python/fetch-download-metadata.py:14]: Oh yeah, I forgot that we marked uv itself as "unmanaged".

**Comment by zanieb** [comment]
Will review this next.

**Comment by charliermarsh** [comment]
Oh I was just gonna merge -- go for it, np.

File: crates/uv/src/lib.rs [hunk 4]
```diff
@@ -67,6 +68,39 @@ async fn run(cli: Cli) -> Result<ExitStatus> {
...
+            Commands::Tool(ToolNamespace {
+                command: ToolCommand::Run(_),
+            }) => false,
+
+            // Supports `--isolated` as its own argument, so we can't warn either way.
+            Commands::Project(command) if matches!(**command, ProjectCommand::Run(_)) => false,
+
+            // `--isolated` moved to `--no-workspace`.
+            Commands::Project(command) if matches!(**command, ProjectCommand::Init(_)) => {
+                warn_user!("The `--isolated` flag is deprecated and will be ignored. Instead, use `--no-config` to prevent uv from discovering configuration files, or `--no-workspace` to prevent uv from adding the initialized project to the containing workspace.");
```
Comment (zanieb) [inline crates/uv/src/lib.rs:84]: Don't you need "files, or **use** `--no-worspace`" for the comma to be correct?

File: crates/uv/src/lib.rs [hunk 5]
```diff
@@ -67,6 +68,39 @@ async fn run(cli: Cli) -> Result<ExitStatus> {
...
+                false
+            }
+
+            // Preview APIs. Ignore `--isolated` and warn.
+            Commands::Project(_) | Commands::Tool(_) | Commands::Python(_) => {
+                warn_user!("The `--isolated` flag is deprecated and will be ignored. Instead, use `--no-config` to prevent uv from discovering configuration files.");
```
Comment (zanieb) [inline crates/uv/src/lib.rs:90]: "will be ignored" is a little ambiguous as to this invocation or in the future. "is deprecated and has no effect"?

**Review by zanieb — APPROVED** [review]
Thought the review request was new for some reason :)

**Pull request merged**